### ISSUE TITLE
Hiking path routes colours and shields

### ISSUE THREAD

**Issue by alensiljak** [organicmaps/organicmaps#7500]
⚠ Have you searched for similar, already existing issues?
Yes, many similar issues but this one is very specific.

**Is your feature request related to a problem? Please describe.**
I would like to see the marked paths clearly indicated with their colour. This is to distinguish between non-marked (and usually non-maintained) paths, as well as between different marked paths.

**Describe the ideal solution**
I would like something the way it is implemented in OsmAnd, see below. The screenshot below is using OSMC colouring scheme, which is the colour of the paint used for markings on trees, stones, shields, etc.

**Describe alternatives you have considered**
Using another app.

<img src="https://github.com/organicmaps/organicmaps/assets/462445/02bea50e-a961-4b99-9c82-61975492b24f" width="300" />

**Additional context**
#2356

**Comment by pastk** [comment]
It was discussed in more detail here:
- https://github.com/organicmaps/organicmaps/issues/2185

**Comment by alensiljak** [comment]
I'd like to make this issue quite specific and limit it to applying the colouring from the Relation. In the screenshot above, some paths are green, others red, blue, yellow. These colours are marked on a Relation: Hiking Path. Often there are also shield colours there.

There's lots of useful information on the linked issue but the title is too broad and generic.

**Comment by vfosnar** [comment]
For example in the Czech Republic as mentioned [here](https://github.com/organicmaps/organicmaps/issues/2185#issuecomment-1305463061) the hiking paths are physically marked, each having its own identifying color, so people can follow them without maps. Having these is a requirement for most ppl in CR as Mapy.cz has a good coverage.

**Comment by alensiljak** [comment]
Correct. In Austria, and some other countries these would be the typical colour trail markers:

![th-3998255998](https://github.com/organicmaps/organicmaps/assets/462445/db360360-4fd2-4650-be13-63e07829555b)

Sometimes they overlap on certain sections and this is an open problem to solve. OsmAnd, for example, picks the first one or a random one, it doesn't even matter.

In former Yugoslavia, these were the markers:

![th-1150491923](https://github.com/organicmaps/organicmaps/assets/462445/4180eca1-cca8-4fd2-921d-b762b9bb11d8)

and they are still in use in the area. In OSM, we mark them with the shield and typically a red colour:

**Comment by fancsali** [comment]
Can't we just grab (and cache) the waymarked-trails slippy tiles?

**Comment by wzbartczak** [comment]
Displaying hiking trails is one of the features that compels me to use other apps. (Well, maybe a separate layer of wikipedia tags as well.) We have quite a few marked trails in Poland, and it's a shame that this data is not used.

**Comment by ikanakova** [comment]
The Czech app Mapy.cz will soon charge for offline (OSM) maps and if Organic Maps had colour marked hiking routes, I'm sure many users would switch from Mapy.cz to Organic Maps. Hiking is very popular in the Czech Republic and Slovakia.

**Comment by fancsali** [comment]
Let's make it happen then – what do we need to achieve that?

**Comment by pastk** [comment]
> Let's make it happen then – what do we need to achieve that?

The main issue for this feature is 
- https://github.com/organicmaps/organicmaps/issues/2185
The current state is - discussions. It needs some committed volunteers to propel its forward :)

**Comment by biodranik** [comment]
User report about hiking path colors in Czech:

---

We have a lots of these paths with different colors and unfortunately other apps displays them sometimes in a wrong way. So I thought that it might be helpful to write to you and compare how different apps tackled it so that Organic Maps would handle it correctly. I am including a few screenshots of the location 49.6745861N, 17.3808139E from different apps:
- [Mapy.cz](http://mapy.cz/) - This app displays the turistic paths in the most accurate way. The paths can have different colors. The color of the path (red, green, blue, yellow) corresponds with the colors of the signs drawn on the trees along the path. There can be also a dashed variation of these colors. The dashed green is used often for educational paths in the nature and so on. These type of paths is drawn with a dashed line in the map and the signs on the trees has a diagonal color line instead of a horizontal one (for example see https://cs.wikipedia.org/wiki/Nau%C4%8Dn%C3%A1_stezka). I'm sorry if I am explaining something that you already know but I think that the Czech Republic (and Slovak Republic also) is specific in having a lot varios types of paths. As for the used maps, [Mapy.cz](http://mapy.cz/) uses their own maps for the Czech Republic but they use OSM maps for the other contries. The app used to be free for years but they recently announced that there would  introduce be subscription soon and some of the features would be blocked in the free version. That is why a lot of users including me are currently looking for alternatives :-)
- Locus - The paths are displayed correctly. However it does not distinguish "solid" and "dashed" paths. This solution is not perfect but still pretty good.
- OsmAnd - It can display the paths but with wrong colors! For example the red path is displayed in blue color and there is only one red sign displayed at to the path on one spot. This is very confusing.

It would be great if displaying these paths could be enabled or disabled in the settings because they are great for walking in the nature but not so useful for other activities.

**Comment by mkyral** [comment]
>  * OsmAnd - It can display the paths but with wrong colors! 
>  For example the red path is displayed in blue color and there is only one red sign displayed at to the path on one spot. This is very confusing.

OsmAnd uses [waymarkedtrails.org](https://hiking.waymarkedtrails.org) style. Colors represents significance of the trail - [key:network](https://wiki.openstreetmap.org/wiki/Key:network)

I't not wrong, it is just different approach more common in western Europe

**Comment by patepelo** [comment]
#Emails&Reviews

1+ User requesting 

Hope we can progress on this soon.

**Comment by vng** [comment]
https://github.com/organicmaps/organicmaps/releases/tag/2025.08.06-4-android

So far, so good! Without shields, but with the primary color from ```symbol/osmc:symbol``` tags.
Feedback is welcome!

**Comment by Tomptez** [comment]
@vng This is amazing, thank you so much!

After testing the routes a bit, I however noticed that there are substantially fewer bike paths available compared to OsmAnd. I am not sure what type of selection is being done compared to OsmAnd, but I feel like it doesn't follow a very clear pattern and many very clearly marked bike paths are missing. Is there any way to include them?

<img width="360" height="760" alt="Image" src="https://github.com/user-attachments/assets/9fd20de7-d2ab-4d9f-abb3-43a382a62d4c" />
<img width="360" height="760" alt="Image" src="https://github.com/user-attachments/assets/ea0e2519-6dcf-4704-bf3d-3dabdc5550ea" />

See screenshot comparison

**Comment by fancsali** [comment]
Quite nice with the new colours feature - would be awesome if we could also add the shields or the names of the routes so we can differentiate between the close-by routes.

Usually they are bunched up, as they are side-branches of each others (e.g. the one marked/named with P being the main route, and P+, PO and PΔ being side-branches leading to crossing points towards other routes, lakes/ponds and peaks (in that order).

All that is to say, it's quite helpful to know, which red is which... ;)

**Comment by alensiljak** [comment]
> So far, so good! Without shields, but with the primary color from `symbol/osmc:symbol` tags. Feedback is welcome!

Thank you so much for the work put into this! I've only noticed this now, in the new Hiking profile that came with the new version. It looks great, as demonstrated in the screenshot above.
Looking forward to eventually getting some shading in the hiking map, making OM much more useful for topographical navigation.

**Comment by vng** [comment]
> Is there any way to include them?

@Tomptez Now OM shows route=bicycle/mtb relations. OsmAnd also highlights highway=cycleway (violet). This is in our nearest plans to highlight regular bicycle roads.

**Comment by vng** [comment]
@fancsali Please, show me a good example from OSM with these route names to debug.

**Comment by eblis** [comment]
Yes, I think i finally found the issue thread for hiking/trail suggestions.

Is anyone looking into adding the symbols (is the symbol the same as shield mentioned above, or is it different)?

In some countries the trails also have a symbol on top of the color, so you can have blue dot, blue line, blue cross .. if you only show the color then you can't tell them apart, it would be very nice to show the color as well.

Mapy does it, I would guess they get the info from OpenStreetMap as well.

<img width="1077" height="546" alt="Image" src="https://github.com/user-attachments/assets/484e54c6-0021-4f2e-843a-7fce5e568e89" />

 Do you need any help with implementing this?

**Comment by eblis** [comment]
I've added some more information in #865, [this comment](https://github.com/orgs/organicmaps/discussions/865#discussioncomment-14317420)

I've extracted all symbols and their counts from OSM and there's also PNGs with all shields available [here](https://wiki.openstreetmap.org/wiki/Trail_markings)

**Comment by tjasz** [comment]
I notice in the new hiking layer, only some route colors are supported. My local area has a park with three waymarked trails:
- [Yellow Loop](https://www.openstreetmap.org/relation/16300081)
- [Dark Blue Loop](https://www.openstreetmap.org/relation/16300080)
- [Light Blue Spur](https://www.openstreetmap.org/relation/16300079)

The relations are tagged in OSM with `colour` = "yellow", "darkblue" and "lightblue" respectively. The color swatches render correctly on the OSM website.

However, on Organic Maps, only the yellow loop is colored yellow. The others are the default colour.
Similarly, on [Waymarked Trails](https://hiking.waymarkedtrails.org/#?map=15.0/47.3097/-122.5303), only the yellow loop color is rendered correctly.

<img height="500" alt="Image" src="https://github.com/user-attachments/assets/636c7754-21b9-41f8-a085-60ea6f1f1542" />

"lightblue" and "darkblue" are recognized [CSS3 colors](https://www.w3schools.com/cssref/css_colors.php). The mappings for them are in [this Python code](https://github.com/organicmaps/organicmaps/blob/127f6277eee30d2e4fb54cfd507fff14b5544632/tools/python/stylesheet/webcolors/webcolors.py#L304), though I don't know enough about Organic Maps architecture yet to know whether or not that is relevant.

Can we get OM to recognize CSS 3 colors in the Hiking style? Or should CSS 3 colors be avoided in OSM tagging in favor of hex codes?

**Comment by vng** [comment]
@tjasz Thanks! Added here (from OSM taginfo top values): https://github.com/organicmaps/organicmaps/pull/11543

**Comment by eblis** [comment]
Is there anything needed by the developers to implement this? 
Or can someone support of I were to attempt to add shield for the Android version?

**Comment by biodranik** [comment]
@eblis let's close this issue if it works with colors, and create another, more detailed issue about shields, with the necessary information about them from osm.org

**Comment by vng** [comment]
Please, don't close. It is about route shields, and the context is fine.
https://github.com/organicmaps/organicmaps/issues/7500#issuecomment-3257320101

**Comment by eblis** [comment]
What's missing then, SVGs for all possible shields? Is there any chance to use PNGs instead, just as an initial test?

I remember that when I last searched the shields I stumbled upon a site that could generate the image based on the values you supplied, but I'm not sure if it generated PNGs or SVGs, I would assume it did PNGs.
Also, could we do the most common images first, just as an initial test and then include (some) more shield values ..

**Comment by mkyral** [comment]
Maybe you mean this one: https://hiking.waymarkedtrails.org/osmc_symbols.html

**Comment by eblis** [comment]
Throwing the links here so I don't have to find them again.

Give a code and get the image:
https://osm.mueschelsoft.de/osmc/
https://hiking.waymarkedtrails.org/osmc_symbols.html

Some (extra) explanations:
https://www.wanderreitkarte.de/info_de.php?lang=en
https://hiking.waymarkedtrails.org/#help-about?map=3.0/34.6/-7.9

**Comment by eblis** [comment]
> Maybe you mean this one: https://hiking.waymarkedtrails.org/osmc_symbols.html

Yes, thank you!
(I didn't check back here until I found the site again 😄  )

Can the SVGs be combined at runtime and given different colors ? Could we have a SVG for a bar with 2 different colors (one for top and bottom, one for middle) and then we just add the colors at runtime when parsing `osmc:symbol` ? And then add a colored hiker SVG in the middle for example? If possible that would make it a lot easier to manage I think, we just need SVGs for the different base options, then we color and combine them at runtime.
